src/java/org/apache/cassandra/config/EncryptionOptions.java

```diff
@@ -147,9 +151,9 @@ public EncryptionOptions()
         ssl_context_factory = new ParameterizedClass("org.apache.cassandra.security.DefaultSslContextFactory",
                                                      new HashMap<>());
         keystore = "conf/.keystore";
-        keystore_password = "cassandra";
+        keystore_password = null;
         truststore = "conf/.truststore";
-        truststore_password = "cassandra";
+        truststore_password = null;
```

Comment:
What's the worst thing that could happen as a result of this change? A `cassandra.yaml` out in the wild that uses "cassandra" as a password implicitly by virtue of not specifying it will suddenly break unless the password is explicitly set to the old default?

That's probably fine in a major release, in any case...

Reply:
@maedhroz 

I do not think that having passwords defaulting to some value is a good idea anyway. A password should be of some complex form, not "cassandra". What value is in that, anyway? It would be quite weird to have a commented password field and than having a certificate / keystore / truststore to have "cassandra" password. That is just strange and I dont think that is used in practice a lot ....

Reply:
While I understand the concern about changing a default value belongs to a major release, I'm not sure that the original issue should wait for 5.0. Any solution shipped with the next 4.1 release is considered better than postpone to 5.0.

Would it possible to leave the default value `cassandra` in place while enabling it to be set null explicitly? IMO, that wouldn't conflict with SemVer policy and could be release with the next 4.1.

Reply:
@maulin-vasavada what do you think about @rtib comment above? I think that would be the most ideal solution which might be shipped in next 4.1 patch release (probably in 4.1.2).

Reply:
Either way, I would probably add an entry in `NEWS.txt` for 5.0 in the "Upgrading" section to explain the default change.

> I do not think that having passwords defaulting to some value is a good idea anyway...That is just strange and I dont think that is used in practice a lot

100% agree, and I don't know why we ever did it that way in the first place :-|

Reply:
If we keep the default value in the EncryptionOptions then it becomes difficult to have tests working for PEM use-cases for the unencrypted key. Hence I would like to keep the null assignment in the code with these changes. If that bumps this upto go in 5.x I am fine with it.

Reply:
I'm fine either way as long as `NEWS.txt` mentions it.

Reply:
@maedhroz Do we need the change in NEWS.txt in this PR or that would be part of the 5.0 release? I can resolve this comment after this.

Reply:
Yes, even if it's in a major release, a change in behavior should be documented there.

Reply:
I am for documenting it and merge it to 4.1 as well as to trunk. I think this patch is for trunk only, currently. I can backport it to 4.1 quite easily, should be nicely backportable.

Reply:
@maedhroz and @smiklosovic I added a text in the News txt file under `Upgrades` for 5.0 for the removal of the default keystore passwords. Please take a look. After that we are good to merge this.